code/modules/projectiles/projectile/magic.dm

```diff
@@ -335,6 +335,57 @@
 			return
 	. = ..()
 
+
+/obj/item/projectile/magic/locker
+	name = "locker bolt"
+	icon_state = "locker"
+	nodamage = TRUE
+	flag = "magic"
+	var/weld = TRUE
+	var/created = FALSE //prevents creation of more then one locker if it has multiple hits
+
+/obj/item/projectile/magic/locker/Bump(atom/A)
```

Comment:
Projectiles have specific function for before hit functionality. That seems better place to put this.

Reply:
I completely forgot about the pngs.
I’ll track down whatever proc it is